src/coreclr/vm/peimagelayout.cpp

```diff
@@ -918,14 +918,14 @@ static bool HavePlaceholderAPI()
             FreeLibrary(hm);
         }
 
-        if (pMapViewOfFile3 == NULL || pVirtualAlloc2 == NULL)
+        if (pMapViewOfFile3 != NULL && pVirtualAlloc2 != NULL)
         {
-            pMapViewOfFile3 = INVALID_ADDRESS_SENTINEL;
-            return false;
+            return true;
         }
     }
 
-    return true;
+    pMapViewOfFile3 = INVALID_ADDRESS_SENTINEL;
+    return false;
```

Comment:
This is now going to return `false` if `pMapViewOfFile3` was set from a previous call now. And then also for all subsequent calls, since it gets set to `INVALID_ADDRESS_SENTINEL`.

Reply:
Yep. Let me revert this part now.